### PR TITLE
fix(metrics): Enforce 'near_' prefix for new metrics

### PR DESCRIPTION
**Pull request by nikurt**
Expect metrics exported by nearcore to have a common prefix. This helps in the following cases:
* Avoids name conflicts with metrics from other systems.
* Helps filter and query metrics
* Makes it easy to understand which binary export a certain metric.

### CODE REVIEW

File: core/o11y/src/metrics.rs [hunk 1]
```diff
@@ -183,3 +195,48 @@ pub fn try_create_histogram_vec(
...
+        "flat_storage_creation_status",
+        "flat_storage_creation_threads_used",
+        "flat_storage_distance_to_head",
+        "flat_storage_head_height",
+    ])
+});
+
+/// Expect metrics exported by nearcore to have a common prefix. This helps in the following cases:
+/// * Avoids name conflicts with metrics from other systems.
+/// * Helps filter and query metrics
```
Comment (akashin) [inline core/o11y/src/metrics.rs:216]: Nit: add punctuation in the end of sentence for consistency with other bullet-points

File: core/o11y/src/metrics.rs [hunk 2]
```diff
@@ -183,3 +195,48 @@ pub fn try_create_histogram_vec(
...
+/// * Makes it easy to understand which binary export a certain metric.
+fn check_metric_near_prefix(name: &str) -> Result<()> {
+    // Some metrics were already introduced that without the desired prefix.
```
Comment (akashin) [inline core/o11y/src/metrics.rs:219]: Nit: "... introduced t̶h̶a̶t̶ without ..."

File: core/o11y/src/metrics.rs [hunk 3]
```diff
@@ -183,3 +195,48 @@ pub fn try_create_histogram_vec(
...
+    // TODO: Consistent metric naming.
```
Comment (akashin) [inline core/o11y/src/metrics.rs:220]: Idea: link a GitHub issue in this "TODO(#XXXX)" to track work on removing the exceptions

**Review by akashin — APPROVED** [review]

**Comment by nagisa** [comment]
FWIW the prometheus [Registry](https://docs.rs/prometheus/latest/prometheus/struct.Registry.html#method.new_custom) type enforces this during the gathering of the metrics, rather than definition of them. At the very least this also saves some memory on not storing the prefix many times over, but more importantly it simply makes the experience of defining metrics more palatable given that you don't need to deal with as many invariants and checks.

**Pull request merged**